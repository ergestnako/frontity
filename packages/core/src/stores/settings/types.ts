type State = {
  frontity: {
    name?: string;
    mode?: string;
  };
};

type Store = {
  state: {
    settings: State;
  };
};

type StoreWithSettings = Store & {
  settings: any;
};

export type Settings = {
  state: State;
  onInitialize?: (store: Store) => void;
  actions?: {
    addSettings: (_, store: StoreWithSettings) => void;
  };
};
