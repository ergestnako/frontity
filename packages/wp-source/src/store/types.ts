import { IConfig, IContext, IAction } from 'overmind';
import { namespaced } from 'overmind/config';
import * as source from '.';

const config = namespaced({ source });

export type Config = IConfig<typeof config>;
export type Context = IContext<Config>;
export type Action<Input = void> = IAction<Config, Input>;


// Action types
export type Fetch = Action<{
  name: string;
  page?: number;
}>;

type Payload = {
  name: string;
  params: { [param: string]: string };
  page: number;
};

export type Handler = (context: Context, payload: Payload) => Promise<void>;

export type Register = Action<{
  pattern: string | RegExp;
  handler: Handler;
}>;

// Effects types

export type IsMatch = (name: string, pattern: string) => boolean;
export type Exec = (name: string, pattern: string) => { [param: string]: string }