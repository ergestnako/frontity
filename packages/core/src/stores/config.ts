import { Namespace } from "@frontity/types/namespace";
import { namespaced } from "overmind/config";
import { Settings } from "./settings/types";

export default ({
  namespaces,
  settings
}: {
  namespaces: { [key: string]: Namespace };
  settings: Settings;
}) =>
  namespaced(
    Object.entries(namespaces).reduce(
      (config, [namespace, module]) => {
        if (module.Store) config[namespace] = module.Store;
        return config;
      },
      { settings }
    )
  );
