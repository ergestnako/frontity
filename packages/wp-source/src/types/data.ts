export type DataMap = {
  [name: string]: Data;
};

export type Data =
  | DataFetching
  | DataArchive
  | DataTaxonomy
  | DataCategory
  | DataTag
  | DataAuthor
  | DataPostTypeArchive
  | DataPostArchive
  | DataDate
  | DataPostType
  | DataPost
  | DataPage
  | DataAttachment
  | Data404;

// DATA FETCHING

export type DataFetching = {
  isFetching: boolean;
  isReady?: boolean;
};

// ARCHIVES

export type DataArchive = DataFetching & {
  isArchive: true;
  page: {
    type: string;
    id: number;
    link: string;
  }[][];
};

export type DataTaxonomy = DataArchive & {
  type: string;
  id: number;
  isTaxonomy: true;
};

export type DataCategory = DataTaxonomy & {
  isCategory: true;
};

export type DataTag = DataTaxonomy & {
  isTag: true;
};

export type DataAuthor = DataArchive & {
  id: number;
  isAuthor: true;
};

export type DataPostTypeArchive = DataFetching & {
  type: string;
  isPostTypeArchive: true;
};

export type DataPostArchive = DataPostTypeArchive & {
  isPostArchive: true;
  isHome: boolean;
  isFrontPage?: boolean;
};

export type DataDate = DataFetching & {
  date: string;
  isDate: true;
};

// POST TYPES

export type DataPostType = DataFetching & {
  type: string;
  id: number;
  isPostType: true;
};

export type DataPost = DataPostType & {
  isPost: true;
};

export type DataPage = DataPostType & {
  isPage: true;
  isFrontPage?: boolean;
};

export type DataAttachment = DataPostType & {
  isAttachment: true;
  isMedia: true;
};

// NOT FOUND (OR ERROR)

export type Data404 = DataFetching & {
  is404: true;
};
