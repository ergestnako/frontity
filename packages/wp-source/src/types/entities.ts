// ENTITIES

export type Entity = Single | Attachment | Taxonomy | Author;

export type Href = {
  href: string;
};

export type Embeddable = {
  embeddable: boolean;
};

export type Term = Href &
  Embeddable & {
    taxonomy: string;
  };

export type Links = {
  collection?: Href[];
  self?: Href[];
  about?: Href[];
  curies?: (Href & {
    name?: string;
    templated?: boolean;
  })[];
  author?: (Href & Embeddable)[];
  replies?: (Href & Embeddable)[];
  "wp:post_type"?: Href[];
  "version-history"?: Href[];
  "wp:attachment"?: Href[];
  "wp:contentmedia"?: (Href & Embeddable)[];
  "wp:featuredmedia"?: (Href & Embeddable)[];
  "wp:term"?: Term[];
};

export type Author = {
  type: "author";
  id: number;
  name: string;
  slug: string;
  description: string;
  link: string;
  meta: any;
  url: string;
  avatar_urls: {
    "24": string;
    "48": string;
    "96": string;
  };
  _links?: Links;
};

export type Taxonomy = {
  _links: Links;
  count: number;
  description: string;
  id: number;
  link: string;
  meta: any;
  name: string;
  parent: number;
  slug: string;
  taxonomy: string;
  type: string;
  yoast_meta?: {
    title: string;
  };
};

export type Embedded = {
  author: Author[];
  "wp:featuredmedia": Attachment[];
  "wp:contentmedia": Attachment[][];
  "wp:term": Taxonomy[];
};

export type Size = {
  file: string;
  height: number;
  mime_type: string; // TODO - set real mime/types
  source_url: string;
  width: number;
};

export type Attachment = {
  _embedded?: Embedded;
  _links: Links;
  alt_text: string;
  author: number;
  caption: string;
  comment_status: "open" | "closed";
  date: string;
  date_gmt: string;
  description: string;
  guid: {
    rendered: string;
  };
  id: number;
  link: string;
  media_details: {
    file: string;
    height: number;
    image_meta: {
      aperture: string;
      camera: string;
      caption: string;
      copyright: string;
      created_timestamp: string;
      credit: string;
      focal_length: string;
      iso: string;
      keywords: [];
      orientation: string;
      shutter_speed: string;
      title: string;
    };
    sizes: {
      full: Size;
      medium: Size;
      thumbnail: Size;
    };
    width: number;
  };
  media_type: string;
  meta: any;
  mime_type: string;
  modified: string;
  modified_gmt: string;
  ping_status: "closed" | "open";
  post: number;
  slug: string;
  source_url: string;
  title: {
    rendered: string;
  };
  type: "attachment";
  yoast_meta: {
    title: string;
  };
};

export type Single = {
  type: string;
  id: number;
  link: string;
  author: number;
  _links: Links;
  categories: number[];
  comment_status: "open" | "closed";
  content: {
    protected: boolean;
    rendered: string;
  };
  content_media: number[];
  date: string;
  date_gmt: string;
  excerpt: {
    protected: false;
    rendered: string;
  };
  featured_media: number;
  guid: {
    rendered: string;
  };
  format: string;
  meta: any;
  modified: string;
  modified_gmt: string;
  ping_status: "open" | "closed";
  slug: string;
  sticky: boolean;
  tags: number[];
  taxonomiesMap: {
    category: number[];
    tag: number[];
  };
  title: {
    rendered: string;
  };
  _embedded: Embedded;
  [other: string]: any; // is this necessary?
};

export type TaxonomyMap = {
  [name: string]: Taxonomy;
};

export type SingleMap = {
  [name: string]: Single;
};

export type AttachmentMap = {
  [name: string]: Attachment;
};

export type AuthorMap = {
  [name: string]: Author;
};