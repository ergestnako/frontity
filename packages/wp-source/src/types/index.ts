export * from "./data";
export * from "./store";
export * from "./entities";
