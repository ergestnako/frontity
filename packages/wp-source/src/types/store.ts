import { IConfig, IContext, IAction, IOnInitialize } from "overmind";
import * as source from "../store";

const settings = {
  actions: {},
  effects: {},
  state: {
    packages: {
      source: {} as Settings
    }
  }
};

export type Config = IConfig<{
  state: {
    source: typeof source.state;
    settings: typeof settings.state;
  };
  actions: {
    source: typeof source.actions;
    settings: typeof settings.actions;
  };
  effects: {
    source: typeof source.effects;
    settings: typeof settings.effects;
  };
}>;

export type Context = IContext<Config>;
export type Action<Input = void> = IAction<Config, Input>;

export type Handler = (
  context: Context,
  payload: {
    name: string;
    params: { [param: string]: any };
    page: number;
  }
) => Promise<void>;

export type OnInitialize = IOnInitialize<Config>;

export type Settings = {
  apiUrl: string;
  isCom: boolean;
};