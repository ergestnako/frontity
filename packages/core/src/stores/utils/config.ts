import { Namespace } from "@frontity/types/namespace";
import { namespaced } from "overmind/config";

export default ({ namespaces }: { namespaces: { [key: string]: Namespace } }) =>
  namespaced(
    Object.entries(namespaces).reduce((config, [namespace, module]) => {
      if (module.Store) config[namespace] = module.Store;
      return config;
    }, {})
  );
