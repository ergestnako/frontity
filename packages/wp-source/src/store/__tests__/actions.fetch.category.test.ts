import { createOvermindMock } from "overmind";
import { namespaced } from "overmind/config";
import { Response, Headers } from "cross-fetch";
import { state, actions, effects } from "..";
import { OnInitialize } from "../../types";

import patterns from "./mocks/patterns";
import categoryJson from "./mocks/category.json";
import postsFromCategoryJson from "./mocks/postsFromCategory.json";

const mockResponse = (body, headers) =>
  new Response(
    JSON.stringify(body),
    headers && {
      headers: new Headers(headers)
    }
  );

describe("fetch", () => {
  test("category for the first time", async () => {
    const { resolver } = effects;
    patterns.forEach(({ pattern, handler }) => {
      resolver.add(pattern, handler);
    });
    // Mock get responses
    const api = {
      get: jest
        .fn()
        .mockResolvedValueOnce(
          mockResponse(categoryJson, {
            "X-WP-Total": 1,
            "X-WP-TotalPages": 1
          })
        )
        .mockResolvedValueOnce(
          mockResponse(postsFromCategoryJson, {
            "X-WP-Total": 10,
            "X-WP-TotalPages": 1
          })
        )
    };
    const config = namespaced({ source: { actions, state, effects } });
    const store = createOvermindMock(config, {
      source: { resolver, api }
    });
    await store.actions.source.fetch({ name: "/category/nature/" });
    expect(api.get.mock.calls).toMatchSnapshot();
    expect(store.state.source.dataMap).toMatchSnapshot();
  });

  test("category that exists in state", async () => {
    const { resolver } = effects;
    patterns.forEach(({ pattern, handler }) => {
      resolver.add(pattern, handler);
    });
    // Mock get responses
    const api = {
      get: jest
        .fn()
        .mockResolvedValueOnce(
          mockResponse(categoryJson, {
            "X-WP-Total": 1,
            "X-WP-TotalPages": 1
          })
        )
        .mockResolvedValueOnce(
          mockResponse(postsFromCategoryJson, {
            "X-WP-Total": 10,
            "X-WP-TotalPages": 1
          })
        )
    };
    const onInitialize: OnInitialize = ctx => {
      ctx.state.source.dataMap["/category/nature/"] = {
        id: 7,
        isArchive: true,
        isCategory: true,
        isFetching: false,
        isReady: true,
        isTaxonomy: true,
        pages: [],
        taxonomy: "category",
        total: 10,
        totalPages: 1
      };
    };
    const config = namespaced({
      source: { actions, state, effects, onInitialize }
    });
    const store = createOvermindMock(config, {
      source: { resolver, api }
    });
    await store.actions.source.fetch({ name: "/category/nature/" });
    expect(api.get).toHaveBeenCalledTimes(2);
  });

  test("try to get a category that doesn't exist", async () => {
    const { resolver } = effects;
    patterns.forEach(({ pattern, handler }) => {
      resolver.add(pattern, handler);
    });
    // Mock get responses
    const api = {
      get: jest.fn().mockResolvedValueOnce(
        mockResponse([], {
          "X-WP-Total": 0,
          "X-WP-TotalPages": 0
        })
      )
    };
    const config = namespaced({ source: { actions, state, effects } });
    const store = createOvermindMock(config, {
      source: { resolver, api }
    });
    await store.actions.source.fetch({ name: "/category/not-a-category/" });
    expect(api.get.mock.calls).toMatchSnapshot();
    expect(store.state.source.dataMap).toMatchSnapshot();
  });

  test("the same category twice gets the data once", async () => {
    const { resolver } = effects;
    patterns.forEach(({ pattern, handler }) => {
      resolver.add(pattern, handler);
    });
    // Mock get responses
    const api = {
      get: jest
        .fn()
        .mockResolvedValueOnce(
          mockResponse(categoryJson, {
            "X-WP-Total": 1,
            "X-WP-TotalPages": 1
          })
        )
        .mockResolvedValueOnce(
          mockResponse(postsFromCategoryJson, {
            "X-WP-Total": 10,
            "X-WP-TotalPages": 1
          })
        )
    };
    const config = namespaced({ source: { actions, state, effects } });
    const store = createOvermindMock(config, {
      source: { resolver, api }
    });
    await store.actions.source.fetch({ name: "/category/nature/" });
    await store.actions.source.fetch({ name: "/category/nature/" });
    expect(api.get).toBeCalledTimes(2);
  });
});
