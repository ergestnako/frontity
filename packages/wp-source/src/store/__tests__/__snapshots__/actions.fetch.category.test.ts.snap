// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch category for the first time 1`] = `
Array [
  Array [
    Object {
      "endpoint": "categories",
      "params": Object {
        "slug": "nature",
      },
    },
  ],
  Array [
    Object {
      "endpoint": "posts",
      "params": Object {
        "_embed": true,
        "categories": 7,
        "page": 1,
        "search": undefined,
      },
    },
  ],
]
`;

exports[`fetch category for the first time 2`] = `
Object {
  "/2016/banff-national-park/": Object {
    "id": 36,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/discovering-iceland/": Object {
    "id": 26,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/iceland/": Object {
    "id": 1,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/kebumen/": Object {
    "id": 42,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/lower-antelope-canyon/": Object {
    "id": 54,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/machu-picchu/": Object {
    "id": 30,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/shinjuku-gyoen-national-garden/": Object {
    "id": 57,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/somewhere-in-romania/": Object {
    "id": 12,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/switzerlands-mountains/": Object {
    "id": 23,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/2016/the-al-hajar-mountains/": Object {
    "id": 33,
    "isFetching": false,
    "isPost": true,
    "isPostType": true,
    "isReady": true,
    "type": "post",
  },
  "/author/davidarenas/": Object {
    "id": 4,
    "isArchive": true,
    "isAuthor": true,
    "isFetching": false,
    "isReady": true,
    "pages": Array [],
  },
  "/author/luisherranz/": Object {
    "id": 1,
    "isArchive": true,
    "isAuthor": true,
    "isFetching": false,
    "isReady": true,
    "pages": Array [],
  },
  "/author/mario/": Object {
    "id": 2,
    "isArchive": true,
    "isAuthor": true,
    "isFetching": false,
    "isReady": true,
    "pages": Array [],
  },
  "/author/orballo/": Object {
    "id": 3,
    "isArchive": true,
    "isAuthor": true,
    "isFetching": false,
    "isReady": true,
    "pages": Array [],
  },
  "/category/architecture/": Object {
    "id": 5,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/cities/": Object {
    "id": 4,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/culture/": Object {
    "id": 6,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/inspiring-stories/": Object {
    "id": 57,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/lifestyle/": Object {
    "id": 59,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/nature/": Object {
    "id": 7,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [
      Array [
        Object {
          "id": 57,
          "link": "https://test.frontity.io/2016/shinjuku-gyoen-national-garden/",
          "type": "post",
        },
        Object {
          "id": 54,
          "link": "https://test.frontity.io/2016/lower-antelope-canyon/",
          "type": "post",
        },
        Object {
          "id": 42,
          "link": "https://test.frontity.io/2016/kebumen/",
          "type": "post",
        },
        Object {
          "id": 36,
          "link": "https://test.frontity.io/2016/banff-national-park/",
          "type": "post",
        },
        Object {
          "id": 33,
          "link": "https://test.frontity.io/2016/the-al-hajar-mountains/",
          "type": "post",
        },
        Object {
          "id": 30,
          "link": "https://test.frontity.io/2016/machu-picchu/",
          "type": "post",
        },
        Object {
          "id": 26,
          "link": "https://test.frontity.io/2016/discovering-iceland/",
          "type": "post",
        },
        Object {
          "id": 23,
          "link": "https://test.frontity.io/2016/switzerlands-mountains/",
          "type": "post",
        },
        Object {
          "id": 12,
          "link": "https://test.frontity.io/2016/somewhere-in-romania/",
          "type": "post",
        },
        Object {
          "id": 1,
          "link": "https://test.frontity.io/2016/iceland/",
          "type": "post",
        },
      ],
    ],
    "taxonomy": "category",
    "total": 10,
    "totalPages": 1,
  },
  "/category/news/": Object {
    "id": 56,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/photography/": Object {
    "id": 3,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/travel/": Object {
    "id": 8,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/category/voices-and-views/": Object {
    "id": 58,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "category",
  },
  "/tag/al-hajar-mountains/": Object {
    "id": 34,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/alps/": Object {
    "id": 41,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/animals/": Object {
    "id": 40,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/antelope-canyon/": Object {
    "id": 18,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/arizona/": Object {
    "id": 19,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/canada/": Object {
    "id": 32,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/culture/": Object {
    "id": 30,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/forests/": Object {
    "id": 50,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/iceland/": Object {
    "id": 9,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/inca/": Object {
    "id": 38,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/indonesia/": Object {
    "id": 29,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/island/": Object {
    "id": 23,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/japan/": Object {
    "id": 15,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/java/": Object {
    "id": 28,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/landscape/": Object {
    "id": 47,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/machu-picchu/": Object {
    "id": 37,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/mountains/": Object {
    "id": 42,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/national-garden/": Object {
    "id": 16,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/national-park/": Object {
    "id": 31,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/nature/": Object {
    "id": 13,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/oman/": Object {
    "id": 35,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/peru/": Object {
    "id": 39,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/rocky-mountains/": Object {
    "id": 33,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/romania/": Object {
    "id": 49,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/switzerland/": Object {
    "id": 43,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/the-carpathian-mountains/": Object {
    "id": 48,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/tokyo/": Object {
    "id": 14,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
  "/tag/united-arab-emirates/": Object {
    "id": 36,
    "isArchive": true,
    "isFetching": false,
    "isReady": true,
    "isTag": true,
    "isTaxonomy": true,
    "pages": Array [],
    "taxonomy": "tag",
  },
}
`;

exports[`fetch try to get a category that doesn't exist 1`] = `
Array [
  Array [
    Object {
      "endpoint": "categories",
      "params": Object {
        "slug": "not-a-category",
      },
    },
  ],
]
`;

exports[`fetch try to get a category that doesn't exist 2`] = `
Object {
  "/category/not-a-category/": Object {
    "is404": true,
    "isFetching": false,
  },
}
`;
